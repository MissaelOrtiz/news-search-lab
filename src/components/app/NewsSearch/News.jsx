/* eslint-disable max-len */
import React from 'react';
import PropTypes from 'prop-types';

const News = ({ author, title, content, urlToImage, name }) => (
  <>
    <img role="img" src={urlToImage} alt={name} />
    <h2>{title}</h2>
    <p> By {author} [via {name}]</p>
    <p>{content}</p>
  </>
);

News.propTypes = {
  author: PropTypes.string,
  title: PropTypes.string,
  content: PropTypes.string.isRequired,
  urlToImage: PropTypes.string,
  name: PropTypes.string
};

export default News;
