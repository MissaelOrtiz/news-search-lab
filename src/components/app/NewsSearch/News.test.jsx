/* eslint-disable max-len */
import React from 'react';
import { render, screen } from '@testing-library/react';
import News from './News';

describe('News', () => {
  it('renders out a single article', () => {
    render(
      <News author="The Technowizard" title="How NOT to fail" content="git gud." urlToImage="https://cdn.discordapp.com/attachments/870470157275910204/882420160567345162/20210624_003711.jpg"/>
    );

    const article = screen.getByText('How NOT to fail');
    expect(article).toMatchSnapshot();
  });
});
