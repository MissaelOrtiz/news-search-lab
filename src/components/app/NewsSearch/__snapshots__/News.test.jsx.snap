// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`News renders out a single article 1`] = `
<h2>
  How NOT to fail
</h2>
`;
